### PR TITLE
DEVELOPER-4474: Fix flaky tests and implement better failure diagnosis 

### PR DESCRIPTION
**Pull request by RubyGemTSL**
[DEVELOPER-4474](https://issues.jboss.org/browse/DEVELOPER-4474): Fix flaky tests and implement better failure diagnosis

### CODE REVIEW

**Comment by LightGuard** [comment]
@RubyGemTSL Looks like something is up with this.

**Comment by RubyGemTSL** [comment]
@LightGuard in what way?

**Comment by RubyGemTSL** [comment]
Do you mean the e2e tests show as failing before the tests complete? I see that, and also the link to the new report is not working.

**Comment by LightGuard** [comment]
@RubyGemTSL It's been two hours and nothing but the build unit tests have run according to the status. Did something die?

**Comment by RubyGemTSL** [comment]
It was building the drupal site for quite a while but once that completed all the checks were green. There was an issue with the details links for e2e's not linking to the correct report and I've just pushed a fix for that.

**Comment by RubyGemTSL** [comment]
Looks like it's finally ready, however Blinkr timed out. I'll do a retest just to be sure.

**Comment by RubyGemTSL** [comment]
retest this please

**Comment by RubyGemTSL** [comment]
@LightGuard I'm going to do more testing on this tomorrow. As you can see the reports look good, however  I plan to create a few branches with this code and then run all three PR at the same time to verify there are no port issues.

**Comment by RubyGemTSL** [comment]
I think this is ready now if you're happy @LightGuard

**Comment by RubyGemTSL** [comment]
@LightGuard I looked into the full screen screenshot then I realised the screenshot on failure functionality is built into [chimp](https://chimp.readme.io/). If you're happy with the current solution I can investigate overriding the chimp screenshot functionality at a later stage?

File: _tests/e2e/config/BrowserManager.js [hunk 1]
```diff
@@ -1,52 +1,42 @@
 "use strict";
+
 class BrowserManager {
 
     createBrowser(browser) {
         if (browser === 'chrome') {
             return this.chromeBrowser('desktop');
-        } else if (browser === 'headless_chrome') {
-            return this.chromeBrowser('headless');
```
[the remark this answers is not in the document]
Comment (RubyGemTSL) [reply _tests/e2e/config/BrowserManager.js]: I removed headless as we use docker selenium, and If developing tests locally I always use a browser, however I might as well add headless back in. No problem.

File: _tests/e2e/config/chimp-local.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    saveScreenshotsToDisk: true,
+    saveScreenshotsToReport: true,
+
+    // - - - - REPORTER - - - -
+    theme: 'bootstrap',
+    jsonFile: `./report/${testProfile}/cucumber-${testProfile}.json`,
+    output: `./report/${testProfile}/cucumber-${testProfile}.html`,
+    reportSuiteAsScenarios: true,
+    launchReport: false
+};
```
[the remark this answers is not in the document]
Comment (RubyGemTSL) [reply _tests/e2e/config/chimp-local.js]: Chimp config serves as a prerequisite for testing based on the environment you wish to run your tests (local, docker, browserstack). it selects the browser to use, where to save reports, screenshots, baseUrl, etc. 

I session on the new automated test setup is long overdue. Apologies for that. I will be doing this in the next couple of weeks.

File: _tests/e2e/config/chromium_devices.json [hunk 3]
```diff
@@ -61,8 +61,8 @@
       "version": "9.1"
     },
     "device": {
-      "name": "Apple iPhone 5",
-      "model": "Apple iPhone 5",
+      "name": "iPhone 5",
+      "model": "iPhone 5",
```
[the remark this answers is not in the document]
Comment (RubyGemTSL) [reply _tests/e2e/config/chromium_devices.json]: Agreed. I'll remove that. we're also still testing on an iPhone6. Maybe we should use a 7 or 8, and possibly a subset of android devices.

**Comment by RubyGemTSL** [comment]
@LightGuard thanks for the review comments. I'll address these over the weekend or Monday.

**Comment by LightGuard** [comment]
retest this please

**Comment by RubyGemTSL** [comment]
retest this please

**Comment by RubyGemTSL** [comment]
retest this please

**Comment by RubyGemTSL** [comment]
@LightGuard I addressed your comments in the last push. Hopefully it's good now.

**Pull request merged**